### PR TITLE
XW-2293 | Productize `Add New Page` button

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/XW-2293

Clean up after XW-2226.

@Wikia/x-wing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L663](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L663): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MINOR [Tracking.js#L586](https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L586): Replace this "switch" statement with "if" statements to increase readability. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1301)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L663](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L663): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MINOR [Tracking.js#L586](https://github.com/Wikia/app/blob/1c2b850ffcf3508243ec07f2c06f48cfd43bf693/skins/oasis/js/Tracking.js#L586): Replace this "switch" statement with "if" statements to increase readability. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1301)

File: skins/oasis/css/core/PageHeader.scss [hunk 1]
```diff
@@ -100,12 +100,10 @@
...
 	}
 
-	.mainpage-add-page {
-		border-right: 1px solid $color-page-border;
+	.wikia-button.createpage {
 		float: right;
-		line-height: 20px;
-		margin-top: 2px;
-		padding: 0 10px;
+		clear: both;
```
Comment (kvas-damian) [inline skins/oasis/css/core/PageHeader.scss:104]: :abc:

File: skins/oasis/modules/templates/PageHeader_Index.php [hunk 2]
```diff
@@ -25,17 +25,12 @@
...
 			</div>
 			<div class="header-column header-tally">
-				<?php if ( !empty( $pageExists ) ) { ?>
-					<div id="PageShareContainer" class="page-share-container">
-						<?= F::app()->renderView( 'PageShare', 'Index' ); ?>
-					</div>
-				<?php } ?>
 				<? if ( !is_null( $tallyMsg ) ): ?>
 					<div class="tally"><?= $tallyMsg ?></div>
+					<a href="/wiki/Special:CreatePage?flow=create-page-contribute-button" class="wikia-button createpage" data-id="createpage">
```
Comment (kvas-damian) [inline skins/oasis/modules/templates/PageHeader_Index.php]: Hardcoded url was meant to be a temporary solution for ABTest. For production ready solution please use `SpecialPage::getTitleFor( $specialPageName )->getLocalURL()`.

Bonus points for reusing existing logic in: https://github.com/Wikia/app/blob/55a5452c79a44f0b49e6886f0352fba87745ecaf/skins/oasis/modules/ContributeMenuController.class.php#L19-L19 which already includes flow tracking parameters.

**Review by kvas-damian — CHANGES_REQUESTED** [review]
Please address feedback

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues
* MAJOR 21 major
* MINOR 7 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [FlowTracking.hooks.php#L32](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L32): Remove the unused function parameters "$out, $skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): Remove the unused function parameters "$page, $user, $text, $summary, $minoredit, $watchThis, $sectionAnchor, $flags". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): This function has 9 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. MAJOR [FlowTracking.hooks.php#L112](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L112): Remove the unused function parameter "$header". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/fcd8720cc89ad90c9e19f52487d3fd775f6c05fa/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: skins/oasis/modules/PageHeaderController.class.php [hunk 3]
```diff
@@ -605,6 +605,24 @@ public function executeHubs( $params ) {
...
+			$this->shouldDisplay = true;
+			$href = SpecialPage::getTitleFor( 'CreatePage' )->getLocalURL();
+
+			wfRunHooks( 'PageHeaderAfterAddNewPageButton', [ &$href ]);
+
+			$this->href = $href;
+		}
+	}
+
+	private function getPageExists() {
```
Comment (kvas-damian) [inline skins/oasis/modules/PageHeaderController.class.php]: what do you think about renaming to `pageExists()`?

Comment (rogatty) [reply skins/oasis/modules/PageHeaderController.class.php]: `doesPageExist()`?

File: skins/oasis/modules/PageHeaderController.class.php [hunk 4]
```diff
@@ -605,6 +605,24 @@ public function executeHubs( $params ) {
...
 
+	public function executeAddNewPageButton() {
+		$this->shouldDisplay = false;
+
+		// That's when we show the number of pages on a wiki
+		if ( $this->getPageExists() && !$this->wg->Title->isSpecial( 'Videos' ) ) {
+			$this->shouldDisplay = true;
+			$href = SpecialPage::getTitleFor( 'CreatePage' )->getLocalURL();
+
+			wfRunHooks( 'PageHeaderAfterAddNewPageButton', [ &$href ]);
```
Comment (kvas-damian) [inline skins/oasis/modules/PageHeaderController.class.php]: spaces ;)

**Review by kvas-damian — APPROVED** [review]

File: skins/oasis/modules/PageHeaderController.class.php (hunk 3 above)
Comment (kvas-damian) [reply skins/oasis/modules/PageHeaderController.class.php]: http://stackoverflow.com/questions/1566745/boolean-method-naming-readability

Comment (weronika) [reply skins/oasis/modules/PageHeaderController.class.php]: `pageExists()` is nicer to read in an `if` :)

File: skins/oasis/js/Tracking.js [hunk 5]
```diff
@@ -219,7 +219,7 @@ jQuery(function ($) {
 
 	/** contribute **/
 
-	$wikiHeader.find('.buttons .contribute').on('mousedown', 'a', function (event) {
+	$wikiHeader.find('.buttons .contribute').add('#WikiaPageHeader').on('mousedown', 'a', function (event) {
```
Comment (weronika) [inline skins/oasis/js/Tracking.js:222]: are you sure it won't catch too many links now?

File: skins/oasis/modules/PageHeaderController.class.php [hunk 6]
```diff
@@ -605,6 +605,24 @@ public function executeHubs( $params ) {
...
+
+			$this->href = $href;
+		}
+	}
+
+	private function pageExists() {
+		return !empty( $this->wg->Title ) && $this->wg->Title->exists();
+	}
+
 	static function onArticleSaveComplete( &$article, &$user, $text, $summary,
```
Comment (WikiaTech) [inline skins/oasis/modules/PageHeaderController.class.php]: MAJOR This function has 11 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues
* MAJOR 21 major
* MINOR 7 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [FlowTracking.hooks.php#L32](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L32): Remove the unused function parameters "$out, $skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): Remove the unused function parameters "$page, $user, $text, $summary, $minoredit, $watchThis, $sectionAnchor, $flags". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): This function has 9 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. MAJOR [FlowTracking.hooks.php#L112](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L112): Remove the unused function parameter "$header". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/b5242adf9c3be0645ae9ae5a9384ac0350db7cf9/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: skins/oasis/js/Tracking.js (hunk 5 above)
Comment (rogatty) [reply skins/oasis/js/Tracking.js:222]: In the callback we filter the links using `data-id` attribute. Also, this is the same selector as in https://github.com/Wikia/app/blob/3ccf876cbf2cce90734ab37d23742a5ede58ccff/skins/oasis/js/Tracking.js#L73

File: skins/oasis/modules/PageHeaderController.class.php (hunk 6 above)
Comment (WikiaTech) [inline skins/oasis/modules/PageHeaderController.class.php:626]: MAJOR This function has 11 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues
* MAJOR 21 major
* MINOR 7 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [FlowTracking.hooks.php#L32](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L32): Remove the unused function parameters "$out, $skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): Remove the unused function parameters "$page, $user, $text, $summary, $minoredit, $watchThis, $sectionAnchor, $flags". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): This function has 9 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. MAJOR [FlowTracking.hooks.php#L112](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L112): Remove the unused function parameter "$header". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/90a0c96de15ce25386df64d4a7e4e0e46dc0e79e/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues
* MAJOR 21 major
* MINOR 7 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [FlowTracking.hooks.php#L32](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L32): Remove the unused function parameters "$out, $skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): Remove the unused function parameters "$page, $user, $text, $summary, $minoredit, $watchThis, $sectionAnchor, $flags". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): This function has 9 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. MAJOR [FlowTracking.hooks.php#L112](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L112): Remove the unused function parameter "$header". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/4fb2e0eb8436827809aacb05d1194cbbc9149c76/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues
- MAJOR 21 major
- MINOR 7 minor
- INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [FlowTracking.hooks.php#L32](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L32): Remove the unused function parameters "$out, $skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
2. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): Remove the unused function parameters "$page, $user, $text, $summary, $minoredit, $watchThis, $sectionAnchor, $flags". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
3. MAJOR [FlowTracking.hooks.php#L50](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L50): This function has 9 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
4. MAJOR [FlowTracking.hooks.php#L112](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L112): Remove the unused function parameter "$header". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
5. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
6. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
7. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
8. MAJOR [Tracking.js#L586](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/skins/oasis/js/Tracking.js#L586): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
9. MAJOR [Tracking.js#L606](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/skins/oasis/js/Tracking.js#L606): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
10. MAJOR [Tracking.js#L648](https://github.com/Wikia/app/blob/87769e74244ba812ea7f9c685ad271272d6f122f/skins/oasis/js/Tracking.js#L648): Function has a complexity of 18 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Pull request merged**